src/modules/ROOT/pages/rfc/record-value-expr/index.adoc

```diff
@@ -0,0 +1,138 @@
+include::../../variables.adoc[]
+
+= Record Value Expressions
+
+Record Value Expressions enables support for expressions to record values and unlock:
+
+ - formula fields
+ - custom sanitizers
+ - custom validators
+ - custom formatting
+
+Introduction of expressions greatly simplifies solutions that required implementation through automation script.
+It brings execution closer to core and within the same process, reducing overhead and improving stability and performance.
+
+Library https://github.com/PaesslerAG/gval[gval] is used for parsing and evaluation of expression.
+This allows to reuse validators and sanitizers on front-end in JavaScript environment.
+
+== Formula field
+
+Formula fields allow dynamic calculation of record values from constants and other values.
+Only constants and values from current objects are supported.
+
+.Example expression for formula field that calculates price with tax:
+[source]
+----
+price * 1.25
+----
+
+This expression assumes there is another field named `price` in record values and multiplies that field with 1.25.
+
+.Example expression for formula field that concatinates 2 string fields
+[source]
+----
+trim(firstName + " " + lastName)
+----
+
+Expression assumes 2 other fields (firstName and lastName) and concatenates them with space between.
+Finally, we trim the result to ensure there result is not prefixed or suffixed with a space in case any of the two fields are empty.
+
+_Later upgrades will allow calculation from referenced records with dependency tracking and cascading updates._
+
+#TBD# How to handle formula change on existing data.
+
+== Sanitizers
+
+Sanitizers aid in record value processing and *cannot raise errors* but *can modify value*.
```

Comment:
On error, it probably falls back to original value and a record in actionlog is created?

Reply:
Actually no. Error raising is in the domain of the validator. Sanitiser just tries to cleanup data for validator and for input. It might change it or not.

This is also the current implementation. We need a good reason to change it.